# Python Prograamming - Assignment 1

**Name:** Aarya Kale
**Roll Number:** MS2512



## Experment 1 - Swapping Two Variables

In [1]:
a = 10
b = 20 

a, b = b, a

print(a)
print(b)

20
10


Initially a has value 10 and b has value 20 

The a, b = b, a swaps the values of a and b . 

Hence the output is a = 20 , b = 10

## Experiment 2 - String Indexing and Slicing 

In [1]:
text = "Python Programming"

print( text[0])
print( text[-1])
print( text[0:6])
print( text[::-1])

P
g
Python
gnimmargorP nohtyP


text[0] - gives first character of string 

text[-1] - using negative index access last character 

text[0:6] - extract characters from index 0 to 6 

text[::-1] - slicing using step value -1 which reverses entire string 

hence output contains indexing slicing and reversing

## Experiment 3 - List Methods

In [5]:
numbers = [10 , 20 , 30 ]

numbers.append(40)

numbers.remove(20)

print (numbers)

[10, 30, 40]


append - used to add new value

remove - used for remove value from list

## Experiment 4 - String Slicing

In [6]:
text = "Programming"

print( text[0:5])
print( text[3 : ])
print( text[:6])
print( text[ : : -1])

Progr
gramming
Progra
gnimmargorP


text[0:5] - extract characters from index 0 to 4

text[3 : ] - extract string from index 3 to end

text[: 6] - extract characters from start to index 5

text[::-1] - reverse entire string


## Experment 5 - Input and Type Conversion

In [7]:
age = input("Enter your age: ")

print(age + age)

age = int(age)

print(age + age)

2222
44


input() - asks for data from user

at first age is stored as string that is why when you enter 22 you get 2222 after string concatenation perform by age + age

int(age) - converts string into integer 

after conversion sge + age performs numerical addition 

## Experment 6 - List Indexing and Slicing 

In [8]:
fruits = ["Apple" , "Banana" , "Mango" , "Orange"]

print(fruits[0])
print(fruits[-1])
print(fruits[1:3])

Apple
Orange
['Banana', 'Mango']


fruits[0] - access first element in list

fruits[-1] - negative indexing to access last element of list 

fruits[1:3] - extracts elements from index 1 to 2

## Experiment 7 - List Mutability

In [9]:
numbers = [10 , 20 , 30]

numbers[1] = 200

print(numbers)

[10, 200, 30]


Initially  list contains [10, 20, 30]

numbers[1] = 200 changes  element at index 1 from 20 to 200

the final list becomes [10, 200, 30]

A list can be modified because lists in Python are **mutable**, which means their elements can be changed after the list is created.

Strings, on the other hand, are **immutable**, which means individual characters of a string cannot be changed after the string is created.

Hence, list elements can be updated directly, while string characters cannot be modified directly.

## Experiment 8 - Formatted Output Using String Repitition

In [10]:
print("=" * 40)
print("WELCOME")
print("=" * 40)

print()

print("*" * 20)
print("Python")
print("*" * 20)

WELCOME

********************
Python
********************


In [ ]:
The expression "=" * 40 repeats the = 40 times and creates a horizontal line.

print("WELCOME") displays the word WELCOME

print() prints an empty line, which adds space between the two sections

The expression "*" * 20 repeats the *  20 times

print("Python") displays the word Python between the two lines of asterisks

## Experiment 8 - Name Reversal

In [11]:
name = input("Enter your name: ")

reverse_name = name[::-1]

print("Reversed name:", reverse_name)

Reversed name: ayraA


input() take name from user

name[::-1] uses step value to reverse the string

reversed string is stored in the variable reverse_name 

## Experiment 9 - Student Information Card

In [12]:
name = input("Enter your name: ")
age = input("Enter your age: ")
roll_number = input("Enter your roll number: ")
course = input("Enter your course: ")

print("=" * 28)
print("STUDENT CARD")
print("=" * 28)

print("Name:", name)
print("Age:", age)
print("Roll Number:", roll_number)
print("Course:", course)

print("=" * 28)

print("Hello " + name + "!")
print("Welcome to the world of Python.")
print('"Python is easy to learn!"')
print("'I'll practice every day.'")
print('"Practice makes perfect"')
print("\\/" * 12)
print("Happy Coding!")

STUDENT CARD
Name: Aarya
Age: 22
Roll Number: MS2512
Course: SCMS
Hello Aarya!
Welcome to the world of Python.
"Python is easy to learn!"
'I'll practice every day.'
"Practice makes perfect"
\/\/\/\/\/\/\/\/\/\/\/\/
Happy Coding!


input() used to accept  student's name, age, roll number, and course from  user

These values are stored in separate variables

The program combines variables, user input, strings, and formatted output to create a simple Student Card.